contracts/deploy/Create3.sol

```diff
@@ -21,11 +30,25 @@ contract CreateDeployer {
     }
 }
 
+/**
+ * @title Create3 Library
+ * @notice This library first deploys the CreateDeployer contract using the CREATE2 opcode and
+ * then utilizes the CreateDeployer contract to deploy a contract with the bytecode parameter
+ * using the CREATE bytecode. This is done to ensure that ensure that the contract bytecode and
+ * constructor arguments of the contract being deployed do not affect its deployed address.
+ */
 library Create3 {
     using ContractAddress for address;
 
     bytes32 internal constant DEPLOYER_BYTECODE_HASH = keccak256(type(CreateDeployer).creationCode);
 
+    /**
+     * @dev Deploys a new contract using the CREATE3 method described above.
+     * @param salt A salt to further randomize the contract address
+     * @param bytecode The bytecode of the contract to be deployed
+     * @return deployed The address of the deployed contract
+     * @custom:requires The bytecode parameter must not be empty.
```

Comment:
I think it's probably better to remove requirements in that case. It would start to clutter the docs if we were to include all requirements enforced by every function, but if we only include some requirements it would be inconsistent.